recirq/time_crystals/dtc_utilities.py

```diff
@@ -0,0 +1,236 @@
+    g_value = sp.Symbol('g')
+
+    # Symbols for random variance (h) and initial state, one per qubit
+    local_fields = sp.symbols('local_field_:' + str(num_qubits))
+    initial_state = sp.symbols('initial_state_:' + str(num_qubits))
+
+    # Symbols used for PhasedFsimGate, one for every qubit pair in the chain
+    thetas = sp.symbols('theta_:' + str(num_qubits - 1))
+    zetas = sp.symbols('zeta_:' + str(num_qubits - 1))
+    chis = sp.symbols('chi_:' + str(num_qubits - 1))
+    gammas = sp.symbols('gamma_:' + str(num_qubits - 1))
+    phis = sp.symbols('phi_:' + str(num_qubits - 1))
+
+    # Initial moment of Y gates, conditioned on initial state
+    initial_operations = cirq.Moment([cirq.Y(qubit) ** initial_state[index] for index, qubit in enumerate(qubits)])
+
+    # First component of U cycle, a moment of XZ gates.
+    sequence_operations = []
+    for index, qubit in enumerate(qubits):
+        sequence_operations.append(cirq.PhasedXZGate(
+                x_exponent=g_value, axis_phase_exponent=0.0,
+                z_exponent=local_fields[index])(qubit))
+
+    # Initialize U cycle
+    u_cycle = [cirq.Moment(sequence_operations)]
+
+    # Second and third components of U cycle, a chain of 2-qubit PhasedFSim gates
+    #   The first component is all the 2-qubit PhasedFSim gates starting on even qubits
+    #   The second component is the 2-qubit gates starting on odd qubits
+    operation_list, other_operation_list = [],[]
+    previous_qubit, previous_index = None, None
+    for index, qubit in enumerate(qubits):
+        if previous_qubit is None:
+            previous_qubit, previous_index = qubit, index
+            continue
+
+        # Add an fsim gate
+        coupling_gate = cirq.ops.PhasedFSimGate(
+            theta=thetas[previous_index],
+            zeta=zetas[previous_index],
+            chi=chis[previous_index],
+            gamma=gammas[previous_index],
+            phi=phis[previous_index]
+        )
+        operation_list.append(coupling_gate.on(previous_qubit, qubit))
+
+        # Swap the operation lists, to avoid two-qubit gate overlap
+        previous_qubit, previous_index = qubit, index
+        operation_list, other_operation_list = other_operation_list, operation_list
+
+    # Add the two components into the U cycle
+    u_cycle.append(cirq.Moment(operation_list))
+    u_cycle.append(cirq.Moment(other_operation_list))
+
+    # Prepare a list of circuits, with n=0,1,2,3 ... cycles many cycles
+    circuit_list = []
+    total_circuit = cirq.Circuit(initial_operations)
+    circuit_list.append(total_circuit.copy())
+    for c in range(cycles):
+        for m in u_cycle:
+            total_circuit.append(m)
+        circuit_list.append(total_circuit.copy())
+
+    return circuit_list
+
+def simulate_dtc_circuit_list(circuit_list: Sequence[cirq.Circuit], param_resolver: cirq.ParamResolver, qubit_order: Sequence[cirq.Qid]) -> np.ndarray:
+    """ Simulate a dtc circuit list for a particular param_resolver
```

Comment:
wrap to 100 characters. consider using an automated formatter like pycharm or `black`